### PR TITLE
Add documentation for since/until

### LINKED ISSUE #1520: Doc for since/until

### PR DESCRIPTION
**Pull request by oldergod**
fixes #1520

### CODE REVIEW

File: wire-library/docs/wire_compiler.md [hunk 1]
```diff
@@ -236,6 +236,38 @@ wire {
 }
 ```
 
+### Version Matching
+
+Members may be declared with `wire.since` and `wire.until` options. For example, these options
```
Comment (Egorand) [inline wire-library/docs/wire_compiler.md:241]: nit: might be great to include a sentence or two with the problem statement, i.e. what are these extensions trying to solve.

File: wire-library/docs/wire_compiler.md [hunk 2]
```diff
@@ -236,6 +236,38 @@ wire {
...
+declare a field `age` that was replaced with `birth_date` in version "5.0":
+
+```proto
+  import "wire/extensions.proto";
+
+  ...
```
Comment (Egorand) [inline wire-library/docs/wire_compiler.md:247]: Maybe include a complete proto?

File: wire-library/docs/wire_compiler.md (hunk 1 above)
Comment (oldergod) [reply wire-library/docs/wire_compiler.md]: fixed

File: wire-library/docs/wire_compiler.md (hunk 2 above)
Comment (oldergod) [reply wire-library/docs/wire_compiler.md]: fixed

**Pull request merged**